middleware/apimiddleware.go

```diff
@@ -229,13 +231,22 @@ func GetBlockHashCtx(r *http.Request) (string, error) {
 
 // GetAddressCtx retrieves the ctxAddress data from the request context. If not
 // set, the return value is an empty string.
-func GetAddressCtx(r *http.Request) string {
-	address, ok := r.Context().Value(CtxAddress).(string)
+func GetAddressCtx(r *http.Request, activeNetParams *chaincfg.Params) (string, error) {
+	addressStr, ok := r.Context().Value(CtxAddress).(string)
 	if !ok {
 		apiLog.Trace("address not set")
-		return ""
+		return "", fmt.Errorf("address not set")
+	}
+	address, err := dcrutil.DecodeAddress(addressStr)
```

Comment:
Seems OK, but as you see in `getAddressInfo` it is run again.

Reply:
yes, future diff will be returning a diff type, if we go that way.